var/spack/repos/builtin/packages/flux-sched/package.py

```diff
@@ -170,6 +171,9 @@ def lua_share_dir(self):
     def lua_lib_dir(self):
         return os.path.join("lib", "lua", str(self.lua_version))
 
+    def setup_build_environment(self, env):
+        env.prepend_path("PYTHON", os.path.join(self.spec["python"].prefix.bin, "python3"))
+
```

Comment:
This looks wrong and should not be needed at all:
```suggestion
```
AFAIK, `"PYTHON"` is not a search path that you can prepend to, but the path to a single executable. `spack` should have already set the build environment so that it is used.

If `spack` would not set this up properly for the build already, the python packages would get the host's python binary and this would show up as soon as someone has an older python on the host than the package needs. This means `spack` already takes care of it, and it not it would be seen in such cases.

Reply:
I don't entirely understand the interplay between the various `setup_x_environment()` functions and the various build systems.  Perhaps you would know the answer to a question we had above that led to this change.  For a CMakePackage, is `setup_run_environment` invoked prior to the `cmake` stage being performed?  This would explain why the change I made in that function initially solved the issue for my user.

Reply:
I don't know exactly, but let your user test the build with `setup_build_envrionment` in the builder class instead.

Reply:
Unfortunately, the user is not in a position to try out a variety of solutions.  We found one that worked for them and the proceeded with their work from there.